### ISSUE TITLE
Access to source code of optic functions

### ISSUE THREAD

**Issue by jcohenadad** [spinalcordtoolbox/spinalcordtoolbox#3438]
## Context

The source code (developed by @sdika) for binaries `isct_spine_detect` and `isct_train_svm` has been unaccessible by the SCT team until now. SCT's team has now access to the code source, but the code needs to remain closed (ie: not shareable via Github).

The source code will notably be useful for:
- ensuring long term support for future OS
- improve workflow of packaging
- <insert your idea here>

The question is: how should we store/manage this codebase?

Related to discussions:
- https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/2669
- https://github.com/spinalcordtoolbox/optic_models/pull/1
- https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/2746

Related to repositories:
- https://github.com/spinalcordtoolbox/optic_models
- https://github.com/spinalcordtoolbox/pmj_models

**Comment by joshuacwnewton** [comment]
> SCT's team has now access to the code source, but the code needs to remain closed (ie: not shareable via Github).

Just to make sure we're 100% clear about @sdika's wishes, some questions:

* Who in the SCT team can have access to the source code? 
   * SCT-specific developers and/or researches, or NeuroPoly as a whole?
   * Depending on the answer, a private GH repo in the @spinalcordtoolbox org may or may not be appropriate, depending on who we give org access to. (Similar conversation to the one we had recently about the Siemens code.)
* Are there any other [conditions or limitations](https://choosealicense.com/licenses/) besides not sharing the source code? (e.g. disclosure of source, modifying it, reproducing it, etc., see the link for a full list)  
* Is there a specific [closed-source license](https://softwareengineering.stackexchange.com/questions/68134/best-existing-license-for-closed-source-code) that the source code has attached to it? Or is this coming more from from word-of-mouth (email, etc.)?

**Comment by joshuacwnewton** [comment]
We may want to also discuss the above questions in the context of https://github.com/neuropoly/onboarding/issues/4, given that I believe we didn't have enough time to go through all of the closed-source repos across NeuroPoly's orgs.

**Comment by jcohenadad** [comment]
> Wonderful news! ❤️
> 
> > SCT's team has now access to the code source, but the code needs to remain closed (ie: not shareable via Github).
> 
> Just to make sure we're 100% clear about @sdika's wishes, some questions:
> 
> * Who in the SCT team can have access to the source code?
>   
>   * SCT-specific developers and/or researches, or NeuroPoly as a whole?

Hum, good question. I would suggest "SCT core developer team". @sdika is that acceptable for you?

>   * Depending on the answer, a private GH repo in the @spinalcordtoolbox org may or may not be appropriate, depending on who we give org access to.

Yes, that was also my thinking

>   * This is similar to the conversation we had recently about the Siemens code from [Audit private repos across all orgs for closed source code that might conflict with base>=Read neuropoly/onboarding#4 (comment)](https://github.com/neuropoly/onboarding/issues/4#issuecomment-867343260)

👍 

> * Are there any other [conditions or limitations](https://choosealicense.com/licenses/) besides not sharing the source code? (e.g. disclosure of source, modifying it, reproducing it, etc., see the link for a more complete set of examples)

Not sure. @sdika can you please fill us in?

> * Is there a specific [closed-source license](https://softwareengineering.stackexchange.com/questions/68134/best-existing-license-for-closed-source-code) that the source code has attached to it? Or is this coming more from from word-of-mouth (email, etc.)?

Hum... not sure. I think they have a custom license. @sdika can confirm.

**Comment by sdika** [comment]
> > Wonderful news! heart
> > > SCT's team has now access to the code source, but the code needs to remain closed (ie: not shareable via Github).
> > 
> > 
> > Just to make sure we're 100% clear about @sdika's wishes, some questions:
> > 
> > * Who in the SCT team can have access to the source code?
> >   
> >   * SCT-specific developers and/or researches, or NeuroPoly as a whole?
> 
> Hum, good question. I would suggest "SCT core developer team". @sdika is that acceptable for you?

yes, that is fine with me, you are in better position to decide who need the access to the code. 
then you have to make sure that people with access to the code will respect the agreement.  

> > * Depending on the answer, a private GH repo in the @spinalcordtoolbox org may or may not be appropriate, depending on who we give org access to.
> 
> Yes, that was also my thinking
> 
> > * This is similar to the conversation we had recently about the Siemens code from [Audit private repos across all orgs for closed source code that might conflict with base>=Read neuropoly/onboarding#4 (comment)](https://github.com/neuropoly/onboarding/issues/4#issuecomment-867343260)
> 
> +1
> 
> > * Are there any other [conditions or limitations](https://choosealicense.com/licenses/) besides not sharing the source code? (e.g. disclosure of source, modifying it, reproducing it, etc., see the link for a more complete set of examples)
> 
> Not sure. @sdika can you please fill us in?

you can modify the source code for maintenance purpose, evolution of dependency (opencv)

> > * Is there a specific [closed-source license](https://softwareengineering.stackexchange.com/questions/68134/best-existing-license-for-closed-source-code) that the source code has attached to it? Or is this coming more from from word-of-mouth (email, etc.)?
> 
> Hum... not sure. I think they have a custom license. @sdika can confirm.

all the details are in the signed agreement

**Comment by jcohenadad** [comment]
> the only written document with all the details is the signed agreement

i’ll upload the signed agreement on the new private repos so it’s clear and transparent for everyone in the team

**Comment by jcohenadad** [comment]
> Hello @jcohenadad, can I have acccess to the `isct_spine_detect` source code?

see: https://forum.spinalcordmri.org/t/where-is-the-c-source-code-for-the-binaries/1122/16?u=jcohenadad
